### ISSUE TITLE
Provide ReadyToRun support for collectible assemblies

### ISSUE THREAD

**Issue by movedoa** [dotnet/runtime#40704]
Currently ReadyToRun is disabled when isCollectible is true in AssemblyLoadContext.
This means that users have to choose between high memory consumption or bad startup performance if they load/unload assemblies as runtime.

In our case its even worse because we load and unload a lot of assemblies an undefined amount of time so our memory consumption would explode without assembly unloading.
This means that currently our performance suffers massively because we cant use r2r assemblies.

Imho there are obvious usecases for this and .NET should support this.

Also this restriction is not very obvious and i couldn't find any documentation that states that this doesn't work.
So the chance is high that many users aren't even aware of this limitation and loose performance.

See also #40683

**Comment by movedoa** [comment]
Is this still on the cards for 6.0?
This is still a big performance problem for us.

**Comment by mangod9** [comment]
This will not make 6.0. There are some OS restrictions to make this work in different assembly contexts.

**Comment by movedoa** [comment]
@mangod9 Are there any updates on this?
This is still a big problem for us.

**Comment by mangod9** [comment]
Would you be able to provide some metrics about what perf issues you are observing? Are these large assemblies where a lot of time is spent in JITing?

**Comment by movedoa** [comment]
Yes the problem definitely is in large assemblies where a lot of time is spent in the JIT.

We did a test of one of the cases we have and the difference is about ~68%.
The time of the first execution/cold start went down from 1100ms to 350ms when using r2r.

We also found out that r2r is blocked when loading an assembly from memory (#74203) which we do so we had to load from a file to do this test.

Fixing this and #74203 would (hopefully) give us a very significant performance boost although it would be conceivable that we always load from a file if there is no other way.

**Comment by agocke** [comment]
Not ready to commit to it yet, but not completely ruling it out either.

**Comment by movedoa** [comment]
@agocke Sorry to be annoying but this is a constant pain point with our customers. 
Is still still in scope for 11?

**Comment by jtschuster** [comment]
Unfortunately, I don't see this making it in 11.
